packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -90,7 +115,26 @@ const action: IRawAction = {
       const rawResponse = await $.http.post('/v1/letters', payload)
       const response = responseSchema.parse(rawResponse.data)
 
-      $.setActionItem({ raw: response })
+      if (
+        !$.step.parameters.shouldGeneratePdf ||
+        !$.flow.hasFileProcessingActions
+      ) {
+        $.setActionItem({
+          raw: response,
+        })
```

Comment:
nit: (did not catch this in the earlier PR)

I'm always a little nervous if we directly expose the API's response into `raw`, because it makes it impedes us from adding metadata later on (due to collision risk).

If we do something like this:
```suggestion
        $.setActionItem({
          raw: {responseData: response},
        })
```

Then we can add metadata like this
```
        $.setActionItem({
          raw: {
            metadata: { /* ... */ },
            responseData: response},
        })
```

If we directly set `raw` = `response` (like current approach), then we would have a problem if the API response also contained `metadata`.

```
        $.setActionItem({
          raw: {
            plumberMetadata: { /* ... */ },
            ...response /* our metadata would be overwritten if response also contained metadata field */
        })
```

But this is quite far fetched. Up to you.

Reply:
wait but im parsing the response data so the return type of response is strictly the 4 attributes? am I missing something here

Reply:
ok i left a comment in the code instead, thanks for bringing this up!